packages/seacas/libraries/ioss/src/unit_tests/UnitTestDynamicTopology.C

```diff
@@ -1106,6 +1108,57 @@ TEST(TestDynamicRead, single_file_simple_topology_modification)
   cleanup_single_file(outFile);
 }
 
+std::tuple<std::string, int, double> read_and_locate_db_state(const OutputParams& params, const double targetTime)
```

Comment:
Might be good to add more tests -- one with all negative times, one with negative and positive.  Also have times in random order `5, 1, 4` to simulate the cyclic time option...

Reply:
Yup, will do

Reply:
I thought the CYCLIC_FILE option only referred to the creation of the filenames based on a modulo reduction. How does it affect the timestep since I thought the timesteps associated with states always had to be monotone increasing. I typically see warning messages if the times are in random order

Reply:
It is posible to use the cyclic option on timesteps within a single file in which case the timesteps would not be monotonically increasing.  There is a warning, but it is allowed.   I think CYCLIC_FILE puts each timestep in a single file and then reuses those files, but there is also a way that a single file will be used and timesteps overwritten.  

For example, if cycle is set to 3 then there would only be 3 timesteps in the file and they would be 1, 2, 3.  Then 1 would be overwritten with 4, 2 with 5, 3 with 6,.  Then 4 with 7, ...

This is in Ioss_DatabaseIO.h:
```
    /*! Typically used for restart output, but can be used for all output...
     * Maximum number of states on the output file.  Overwrite the existing
     * steps in a cyclic manner once exceed this count.  Note that this breaks
     * the convention that times be monotonically increasing on an exodusII file.
     * Used by derived classes if they support this capability...
     */
    mutable int cycleCount{0};
```